### ISSUE TITLE
With expressions incorrectly request prefixing local calls with `this`

### ISSUE THREAD

**Issue by MagusZero** [DotNetAnalyzers/StyleCopAnalyzers#4078]
I've never posted an issue here, so I apologize if I have anything wrong, but this has been bothering me for over a year.

C# 9 introduced the `with` expression, which can be used to great effect to copy immutable records with changes to one or more property.

The syntax is along the lines of:

```csharp
var x = y with
{
    Name = "New Name",
};
```

Ever since this feature was added, that `Name` property has been incorrectly flagged by the stylecop analyzer rule SA1101 as a local call, and therefore wants the `this` prefix. This is clearly invalid and does not even compile. This rule can be disabled, but only globally, and I generally agree with the rule in places where it compiles.

**Comment by MagusZero** [comment]
As an aside, you cannot even reasonably pragma it, because that would cause the following local function use to not have the rule enforced, meaning `SomeFunction` could leave out the `this`.

```csharp
var x = y with
{
    SpecificDetail = this.SomeFunction(y.SpecificDetail),
};
```
